sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/aggregate/ApproximatePercentile.scala

```diff
@@ -267,35 +263,40 @@ object ApproximatePercentile {
   // The default relative error can be deduced by defaultError = 1.0 / DEFAULT_PERCENTILE_ACCURACY
   val DEFAULT_PERCENTILE_ACCURACY: Int = 10000
 
+  def nextPowOf2(relativeError: Double): Int = {
+    val baseK = DoublesSketch.getKFromEpsilon(relativeError, true)
+    if (baseK == 1 || (baseK & (baseK - 1)) == 0) {
```

Comment:
datasketches's K seems has a low bound of 2: https://github.com/apache/datasketches-java/blob/4d1b5b739d675e3a2abd2b52a2d39528c0c53052/src/main/java/org/apache/datasketches/quantiles/ClassicUtil.java#L51